### PR TITLE
chore: Update parseAWSExports to v6 config

### PR DESCRIPTION
**Pull request by cshfang**
#### Description of changes

Update `parseAWSExports` to output v6 config

#### Description of how you validated changes
`yarn test`

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#steps-towards-contributions)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/core/src/parseAWSExports.ts [hunk 1]
```diff
@@ -1,91 +1,101 @@
...
-import { ConsoleLogger as Logger } from './Logger';
-
-const logger = new Logger('Parser');
+import { OAuthConfig } from './singleton/Auth/types';
+import { ResourcesConfig } from './singleton/types';
 
 /**
- * This utility generates an `AmplifyConfig` object from an `aws-exports.js` file generated by the Amplify CLI.
- * 
+ * This utility converts the `aws-exports.js` file generated by the Amplify CLI into a `ResourcesConfig` object
```
Comment (HuiSF) [inline packages/core/src/parseAWSExports.ts:7]: nit: 
```suggestion
 * This utility converts the `aws-exports.js` file generated by the Amplify CLI into a {@link ResourcesConfig} object
```

File: packages/core/src/parseAWSExports.ts [hunk 2]
```diff
@@ -1,91 +1,101 @@
...
+ * consumable by Amplify.
+ *
  * @param config A configuration object from `aws-exports.js`.
- * 
- * @returns An AmplifyConfig object.
+ *
+ * @returns A `ResourcesConfig` object.
```
Comment (HuiSF) [inline packages/core/src/parseAWSExports.ts:12]: nit:
```suggestion
 * @returns A {@link ResourcesConfig} object.
```

File: packages/core/src/parseAWSExports.ts [hunk 3]
```diff
@@ -1,91 +1,101 @@
...
-	if (config['geo']) {
-		amplifyConfig.Geo = Object.assign({}, config.geo);
-		if (config.geo['amazon_location_service']) {
-			amplifyConfig.Geo = {
-				AmazonLocationService: config.geo['amazon_location_service'],
+	if (geo) {
+		const { amazon_location_service } = geo;
+		if (amazon_location_service) {
+			(amplifyConfig as any).Geo = {
+				AmazonLocationService: amazon_location_service,
```
Comment (AllanZhengYP) [inline packages/core/src/parseAWSExports.ts:74]: I think we agreed to remove any `Amazon` and `Service` in the key name. cc @elorzafe

**Review by AllanZhengYP — COMMENTED** [review]
I'm not sure about the Geo. otherwise it looks✅

**Review by AllanZhengYP — DISMISSED** [review]

**Review by HuiSF — APPROVED** [review]
Thanks

File: packages/core/src/parseAWSExports.ts (hunk 3 above)
Comment (cshfang) [reply packages/core/src/parseAWSExports.ts:74]: This is definitely going to change - Geo isn't even a valid property right now on the config. I just wanted to keep the status quo for Geo at this time

**Review by HuiSF — DISMISSED** [review]

**Review by AllanZhengYP — APPROVED** [review]

**Review by HuiSF — APPROVED** [review]

**Review by jimblanc — APPROVED** [review]
Thanks! Looks good

**Pull request merged**